spec.bs

```diff
@@ -989,6 +997,43 @@ With the above infrastructure in place, we can actually fire and handle the {{Ap
     :: "{{AppHistoryNavigationType/push}}"
 </div>
 
+<h3 id="navigate-event-traversal-patches">History traversal updates</h3>
+
+<div algorithm="traverse the history by a delta">
+  The <a spec="HTML">traverse the history by a delta</a> algorithm will be totally re-written as part of <a href="https://github.com/whatwg/html/pull/6315">the session history rewrite</a>. Here we reproduce the final version of the algorithm, after both that rewrite and with appropriate app history updates. Recall that per [[#user-initiated-patches]] we have introduced a |userInvolvement| argument, so the arguments are now |delta|, |source browsing context|, and |userInvolvement|.
+
+  1. Let |traversable| be |source browsing context|'s [=browsing context/containing navigable=]'s [=navigable/traversable navigable=].
+  1. Let |initiatorOrigin| be |source browsing context|'s [=active document=]'s [=Document/origin=].
+  1. [=parallel queue/enqueue steps|Enqueue the following steps=] to |traversable|'s [=traversable navigable/session history traversal queue=]:
+    1. Let |allSteps| be the result of [=traversable navigable/getting all history steps=] for |traversable|.
+    1. Let |targetStep| be TODO compute using [=traversable navigable/current session history step=], |allSteps|, and |delta|. Will be failure if there is no step |delta| away from the current one.
```

Comment:
Something like:

```suggestion
    1. Let |currentStepIndex| be the index of the [=traversable navigable/current session history step=] within |allSteps|.
    1. Assert: |currentStepIndex| is not failure.
    1. Let |targetStepIndex| be |currentStepIndex| plus |delta|.
    1. If |allSteps|[|targetStepIndex|] does not [=set/exist=], then return.
    1. [=Apply the history step=] |allSteps|[|targetStepIndex|] to |traversable| with true, <var ignore>step</var>, |initiatorOrigin|, and |userInvolvement|.
```

Although there's no "index of". Maybe that should be in infra?

Reply:
I think "index of" is relatively self-explanatory so I'm not too bothered about it not existing in Infra. Thanks, this is perfect!